src/components/elements/table/RestockPurchasesTable.tsx

```diff
@@ -0,0 +1,220 @@
+  label: string;
+  numeric: boolean;
+}
+
+const headCells: Array<HeadCell> = [
+  {columnName: 'id', numeric: false, label: 'Order #'},
+  {columnName: 'purchaseDate', numeric: false, label: 'Purchase Date'},
+  {columnName: 'status', numeric: false, label: 'Status'},
+  {columnName: 'type', numeric: false, label: 'Type'},
+  {columnName: 'totalPrice', numeric: true, label: 'Total Price ($)'},
+];
+
+type EnhancedTableProps = {
+  classes: ReturnType<typeof useStyles>;
+  onRequestSort: (event: React.MouseEvent<HTMLSpanElement>, property: keyof RestockDataEntry) => void;
+  order: Order;
+  orderBy: string;
+}
+
+const EnhancedTableHead = (props: EnhancedTableProps) => {
+  const {classes, order, orderBy, onRequestSort} = props;
+  const handleSort = (property: keyof RestockDataEntry) => (event: React.MouseEvent<HTMLSpanElement>) => {
+    onRequestSort(event, property);
+  };
+
+  const tableClasses = useStyles();
+
+  return (
+    <TableHead>
+      <TableRow className={tableClasses.tableRow}>
+        {headCells.map((headCell) =>
+          <TableCell
+            key={headCell.columnName}
+            sortDirection={orderBy === headCell.columnName ? order : false}
+          >
+            <TableSortLabel
+              active={orderBy === headCell.columnName}
+              direction={orderBy === headCell.columnName ? order : 'asc'}
+              onClick={handleSort(headCell.columnName)}
+            >
+              {headCell.label}
+              {orderBy === headCell.columnName ? (
+                <span className={classes.visuallyHidden}>
+                  {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
+                </span>
+              ) : null}
+            </TableSortLabel>
+          </TableCell>,
+        )}
+        <TableCell>Purchase Info</TableCell>
+      </TableRow>
+    </TableHead>
+  );
+};
+
+const RestockPurchasesTable = () => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const {restockData} = useRestockSelector();
+
+  const [order, setOrder] = React.useState<Order>('asc');
+  const [orderBy, setOrderBy] = React.useState<keyof RestockDataEntry>('id');
+  const [page, setPage] = React.useState(0);
+  const [rowsPerPage, setRowsPerPage] = React.useState(10);
+
+  const getData = () => {
+    dispatch(restockDispatchers.fetchRestockData());
+  };
+
+  useEffect(() => {
+    getData();
+  }, []);
+
+  const handleRequestSort = (event: React.MouseEvent<unknown>, property: keyof RestockDataEntry) => {
+    const isAsc = orderBy === property && order === 'asc';
+    setOrder(isAsc ? 'desc' : 'asc');
+    setOrderBy(property);
+  };
+
+  const handleChangePage = (event: React.MouseEvent<HTMLButtonElement, MouseEvent> | null, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  if (!restockData.length) {
```

Comment:
`if (!restockData)` should suffice

Reply:
I needed it to be like that. It doesn't work your way. I've tried multiple times.